code/modules/integrated_electronics/core/saved_circuits.dm

```diff
@@ -232,6 +232,9 @@
 
 	var/error
 
+	//Check for bad inputs
+	var/regex/bad_regex = new("\[^\\w{}:\",.\\-*&+ =\'\\(\\)\\\[\\\]#\]", "gi")
```

Comment:
This regex should be generated once on subsystem init and stored on the subsystem.